packages/tailwindcss/src/candidate.ts

```diff
@@ -510,21 +514,23 @@ export function parseVariant(variant: string, designSystem: DesignSystem): Varia
 
     let selector = decodeArbitraryValue(variant.slice(1, -1))
 
-    if (selector[0] !== '@') {
-      // Ensure `&` is always present by wrapping the selector in `&:is(…)`
-      //
-      // E.g.:
-      //
-      // - `[p]:flex`
-      if (!selector.includes('&')) {
-        selector = `&:is(${selector})`
-      }
+    let relative = selector[0] === '>' || selector[0] === '+' || selector[0] === '~'
+
+    // Ensure `&` is always present by wrapping the selector in `&:is(…)`,
+    // unless it's a relative selector like `> img`.
+    //
+    // E.g.:
+    //
+    // - `[p]:flex`
+    if (selector[0] !== '@' && !selector.includes('&') && !relative) {
```

Comment:
Big nitpick: teeny tiny micro optimization, but feels conceptually more correct to me:
```suggestion
    if (!relative && selector[0] !== '@' && !selector.includes('&')) {
```

I think it's worth checking for the simple boolean first, before checking anything related to the string.

Reply:
Agree!